x-pack/legacy/plugins/reporting/export_types/common/lib/screenshots/observable.test.ts

```diff
@@ -0,0 +1,194 @@
+              "description": "Default ",
+              "title": "Default Mock Title",
+            },
+          ],
+          "timeRange": "Default GetTimeRange Result",
+        },
+      ]
+    `);
+  });
+
+  it('pipelines multiple urls into', async () => {
+    // mock implementations
+    const mockScreenshot = jest.fn().mockImplementation((item: ElementsPositionAndAttribute) => {
+      return Promise.resolve(`allyourBase64 screenshots`);
+    });
+
+    // mocks
+    mockBrowserDriverFactory = await createMockBrowserDriverFactory(logger, {
+      screenshot: mockScreenshot,
+    });
+
+    // test
+    const getScreenshots$ = screenshotsObservableFactory(__LEGACY, mockBrowserDriverFactory);
+    const result = await getScreenshots$({
+      logger,
+      urls: ['/welcome/home/start/index2.htm', '/welcome/home/start/index.php3?page=./home.php'],
+      conditionalHeaders: {} as ConditionalHeaders,
+      layout: mockLayout,
+      browserTimezone: 'UTC',
+    }).toPromise();
+
+    expect(result).toMatchInlineSnapshot(`
+      Array [
+        Object {
+          "screenshots": Array [
+            Object {
+              "base64EncodedData": "allyourBase64 screenshots",
+              "description": "Default ",
+              "title": "Default Mock Title",
+            },
+          ],
+          "timeRange": "Default GetTimeRange Result",
+        },
+        Object {
+          "screenshots": Array [
+            Object {
+              "base64EncodedData": "allyourBase64 screenshots",
+              "description": "Default ",
+              "title": "Default Mock Title",
+            },
+          ],
+          "timeRange": "Default GetTimeRange Result",
+        },
+      ]
+    `);
+  });
+
+  describe('error handling', () => {
+    it('fails if error toast message is found', async () => {
+      // mock implementations
+      const mockWaitForSelector = jest.fn().mockImplementation((selectorArg: string) => {
+        const { toastHeader } = mockSelectors;
+        if (selectorArg === toastHeader) {
+          return Promise.resolve(true);
+        }
+        // make the error toast message get found before anything else
+        return Rx.interval(100).toPromise();
+      });
+
+      // mocks
+      mockBrowserDriverFactory = await createMockBrowserDriverFactory(logger, {
+        waitForSelector: mockWaitForSelector,
+      });
+
+      // test
+      const getScreenshots$ = screenshotsObservableFactory(__LEGACY, mockBrowserDriverFactory);
+      const getScreenshot = async () => {
+        return await getScreenshots$({
+          logger,
+          urls: [
+            '/welcome/home/start/index2.htm',
+            '/welcome/home/start/index.php3?page=./home.php3',
+          ],
+          conditionalHeaders: {} as ConditionalHeaders,
+          layout: mockLayout,
+          browserTimezone: 'UTC',
+        }).toPromise();
+      };
+
+      await expect(getScreenshot()).rejects.toMatchInlineSnapshot(
```

Comment:
NOTE: This part of assertions will change when https://github.com/elastic/kibana/issues/57008 is implemented